### PR TITLE
[ntuple] Fix race in RClusterPool destruction

### PR DESCRIPTION
**Pull request by jblomer**
Ensures that the cluster prefetch thread flushes all queued requests before processing the termination signal.

### CODE REVIEW

File: tree/ntuple/v7/src/RClusterPool.cxx [hunk 1]
```diff
@@ -115,7 +116,8 @@ void ROOT::Experimental::Detail::RClusterPool::ExecReadClusters()
          while (!fReadQueue.empty()) {
             if (fReadQueue.front().fClusterKey.fClusterId == kInvalidDescriptorId) {
                fReadQueue.pop();
-               return;
+               terminate = true;
+               break;
```
Comment (pcanal) [inline tree/ntuple/v7/src/RClusterPool.cxx:120]: In this error case, should we still call "LoadClusters" on the others clusters or should we just discard them all?

Comment (jblomer) [reply tree/ntuple/v7/src/RClusterPool.cxx:120]: We should still call `LoadClusters` on the ones that were in the pipe before the termination marker. @eguiraud pointed the issue out from the [unit tests](https://lcgapp-services.cern.ch/root-jenkins/job/roottest-pullrequests-build/9815/testReport/projectroot.tree.ntuple.v7/test/gtest_tree_ntuple_v7_test_ntuple_cluster)

Comment (jblomer) [reply tree/ntuple/v7/src/RClusterPool.cxx:120]: On second thought, I think actually we shouldn't.  Why would we need to continue prefetching if the cluster pool gets destructed.  That makes testing a bit more complicated, I need to give it one more round of thought.  Thanks for the comment!

**Comment by jblomer** [comment]
@pcanal @eguiraud Second attempt at this issue.

**Comment by jblomer** [comment]
@phsft-bot build

**Comment by jblomer** [comment]
@phsft-bot build

**Review by jalopezg-r00t — APPROVED** [review]
LGTM!  :+1:

**Pull request merged**